airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -701,9 +703,15 @@ def _merge_stream_slicers(self, model: DeclarativeStreamModel, config: Config) -
             )
         elif model.incremental_sync:
             return self._create_component_from_model(model=model.incremental_sync, config=config) if model.incremental_sync else None
+        elif hasattr(model.retriever, "paginator") and model.retriever.paginator and stream_slicer:
```

Comment:
because substream RFR is only checkpointing based on the complete sync of a parent partition, I don't think the paginator is actually a requirement of RFR for substreams.

Could we simplify this to just `elif stream_slicer`? At this point in the conditional block, we should know that:

- We are non a per-partition (list or substream) + incremental stream
- We are not an incremental stream 

So then at this point we only need to decide if this is a substream (and can therefore use RFR substreams) or if it is a regular stream that has a paginator (can use RFR) or does not have a paginator (cant use RFR)

Reply:
Based on the points above, it should be:
```python
elif not model.incremental_sync and stream_slicer:
    ....
```
That matches the: 
- non-incremental stream
- has the `stream_slicer`
- we don't care about having `paginator`

@brianjlai Please verify and confim. 

The only test that failed after changing to `conditional statement` above, is:
```bash
poetry run pytest unit_tests/sources/declarative/parsers/test_model_to_component_factory.py -k 'test_merge_incremental_and_partition_router' -s
```

In this case: `id="test_create_simple_retriever_with_partition_router"`, which now expects the `PerPartitionCursor` instead of `ListPartitionRouter`. So changed that to the correct `expected_type` makes the test to `pass green`.

Reply:
I think we're mostly on the same thought process, but due to the previous condition `if model.incremental_sync and stream_slicer` that is executed before our new condition, we should already be able to know that the stream is not incremental based on line 696 of our code.

That was the main reason I was saying that we actually don't need to check if this is incremental because `model.incremental_sync and stream_slicer` and `not model.incremental_sync and stream_slicer` are mutually exclusive